File: app/src/main/java/noahzu/github/io/gank/HistoryGank/HistoryGankContract.java
package noahzu.github.io.gank.HistoryGank;

import java.util.List;

import noahzu.github.io.gank.Base.BasePresenter;
import noahzu.github.io.gank.Base.BaseView;
import noahzu.github.io.gank.Data.entity.Gank;
import noahzu.github.io.gank.Data.entity.HistoryGankResult;

/**
 * Created by Administrator on 2016/7/10.
 */
public class HistoryGankContract {
    interface View extends BaseView<Presenter>{
        void showMessage(String msg);
        void showLoading();
        void hideLoading();
        void showGanks(List<HistoryGankResult.PreviewGank> ganks);
        void showGankDetails(HistoryGankResult.PreviewGank gank);
        void showPicture(HistoryGankResult.PreviewGank gank);
        int getCurrentPage();
    }

    interface Presenter extends BasePresenter{

    }
}

File: app/src/main/java/noahzu/github/io/gank/HistoryGank/HistoryGankContract.kt
package noahzu.github.io.gank.HistoryGank

import noahzu.github.io.gank.Base.BasePresenter
import noahzu.github.io.gank.Base.BaseView
import noahzu.github.io.gank.Data.entity.Gank
import noahzu.github.io.gank.Data.entity.HistoryGankResult

/**
 * Created by Administrator on 2016/7/10.
 */
class HistoryGankContract {
    internal interface View : BaseView<Presenter> {
        fun showGanks(ganks: List<HistoryGankResult.PreviewGank>)
        fun addGanks(ganks: List<HistoryGankResult.PreviewGank>)
        fun showGankDetails(gank: HistoryGankResult.PreviewGank)
        fun showPicture(gank: HistoryGankResult.PreviewGank)
        val currentPage: Int
        fun startRefresh()
        fun stopRefresh()
        fun startLoading()
        fun stopLoadingMore()
    }

    internal interface Presenter : BasePresenter {
        fun loadGanks()
        fun loadMoreGanks()
    }
}
